chart/templates/secret.yaml

```diff
@@ -15,7 +15,11 @@ kind: Secret
 metadata:
```

Comment:
What happens in the case where you do helm upgrade with an existing bootstrap-secret? Looks like we take the bootstrap password value from the existing secret, then update the same secret with the same bootstrap password value we took which seems redundant.

Reply:
Before this PR, the secret got deleted (see root cause comment in the issue). Without it being a hook, it will fail because of the existing secret. With it being a hook, it will recreate but not fail which is the best outcome for the user.